### ISSUE TITLE
[BUG] Epoch chart is rendered as a filled rectangle instead of a line

### ISSUE THREAD

**Issue by Felrix** [gradio-app/trackio#543]
Hi, I noticed an issue with how the `epoch` metric is displayed in the chart UI.

# Expected behavior

The `epoch` chart should be rendered as a normal line chart, similar to other scalar metrics. Since `epoch` is logged as a scalar value over time, I expected it to appear as a line or points.

# Actual behavior

The `epoch` chart is rendered as a large filled rectangle instead of a line and trackio only record last epoch.
However, the per-step `train/loss` chart is displayed correctly.

# Screenshot

<img width="1130" height="738" alt="Image" src="https://github.com/user-attachments/assets/b96a2fa5-9d9b-458f-9971-95cc4ab2c8f8" />

# Code

```python
for epoch in epochs:
    total_loss = 0.0

    for X, y in dataloader:
        loss = loss_fn(model(X), y)
        total_loss += loss.item()
        trackio.log({"train/loss": loss.item()}, step=epoch)

    avg_loss = total_loss / len(dataloader)
    metrics = {
        "epoch": epoch,
        "train/avg_loss": avg_loss,
    }
    trackio.log(metrics, step=epoch)
```

**Comment by abidlabs** [comment]
Thanks for the report, I dug in and you're hitting a real bug on our side, but there's also a tweak to your logging that's worth knowing about.

**The bug:** when smoothing is on (it's on by default), our smoothing code in `dataProcessing.js` walks a moving window over *every* row of the run and writes a value for every metric, even on rows that didn't originally log that metric. `epoch` is only present on one row per epoch (the rest of the rows in that step are your per-batch `train/loss` calls), so smoothing fabricates an `epoch` value for all of them. The chart then draws one point per row instead of one per epoch — many duplicate x-values stacked at each step, which Vega renders as that vertical fill you're seeing. The data in the DB is fine; every epoch is recorded. We'll get a fix out.

**The logging tweak:** you're calling `trackio.log({"train/loss": ...}, step=epoch)` inside the inner batch loop, so every batch in epoch *k* gets pinned to `step=k`. That means your `train/loss` chart isn't actually showing per-batch loss over time — it's hundreds of points piled on top of each other at each integer step. It happens to look reasonable because the column is dense, but it's not what you probably want.

A more typical pattern:

```python
for epoch in epochs:
    for X, y in dataloader:
        loss = loss_fn(model(X), y)
        trackio.log({"train/loss": loss.item(), "epoch": epoch})  # no step= → auto-increments
    trackio.log({"train/avg_loss": avg_loss, "epoch": epoch})
```

Letting `step` auto-increment gives you a real per-batch loss curve, and tagging `epoch` on every log lets you switch the x-axis to `epoch` in the dashboard if you want to.

Either way, the rectangle is on us, thanks for filing it.
